### ISSUE TITLE
[RLLIB] PPO Gradient Removed for Value Estimation

### ISSUE THREAD

**Issue by dkupsh** [ray-project/ray#51748]
### What happened + What you expected to happen

When using PPO and a custom RLModule, gradient information gets removed for when its computing the value function directly. Thus, if you were to store the value into the batch when doing the forward method and then return that value during the compute_values method, it will cause an error as pytorch recieves no gradients. 

In my model, its better to compute both the logits and value at the same time (not seperately). So, this means that we must either recompute everything during the value function or it causes an error

### Versions / Dependencies

Ray Master Branch

### Reproduction script

```

@override(ValueFunctionAPI)
    def compute_values(
        self,
        batch: Dict[str, Any],
        embeddings: Optional[Any] = None,
    ) -> TensorType:
        return batch[VALUE]
```

where batch[VALUE] is computed within the forward_train method

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by simonsays1980** [comment]
@dkupsh thanks for raising this issue. I am afraid that this is not possible with the default PPO learner. For this you would need to write your own PPO learner following the default learner [here](https://github.com/ray-project/ray/blob/caf3680bf251ae2b393952b44b290b6f877c57f1/rllib/algorithms/ppo/torch/ppo_torch_learner.py#L28).
